buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -650,6 +664,24 @@ public ByteBuf setZero(int index, int length) {
     }
 
     @Override
+    public int setCharSequence(int index, CharSequence sequence, Charset charset) {
+        if (charset.equals(CharsetUtil.UTF_8)) {
+            int len = sequence.length();
```

Comment:
@Scottmitch I kind of not like to add these method as this make the interface even broader :(